File: hello-java/app/controllers/Hello2.java
package controllers;

import java.io.UnsupportedEncodingException;
import java.net.URLDecoder;

import play.mvc.Controller;
import play.mvc.Result;

public class Hello2 extends Controller {
  
	/**
	 * views.hello3 に引数として hello world を渡して表示する。
	 */
	public static Result hello(String msg) {
		return ok(views.html.hello3.render(msg));
	}
  
	/**
	 * views.hello3 に引数として hello world を渡して表示する。
	 */
	public static Result helloWithDecode(String msg) throws UnsupportedEncodingException {
		return ok(views.html.hello3.render(URLDecoder.decode(msg, "utf-8")));
	}
  
}

File: hello-scala/app/controllers/Hello2.scala
package controllers

import play.api.mvc.Controller
import play.api.mvc.Action
import java.net.URLDecoder

object Hello2 extends Controller {

  /**
   * routes から引数として hello world を受け取って表示する。
   */
  def hello(msg: String) = Action {
    Ok(views.html.hello3.render(msg))
  }

  /**
   * routes から引数として hello world を受け取って表示する。
   * encode されたスペース（%20）が decode されなかったので明示的に decode。
   */
  def helloWithDecode(msg: String) = Action {
    Ok(views.html.hello3.render(URLDecoder.decode(msg, "utf-8")))
  }
}